plugin/db/util/driverutil.go

```diff
@@ -183,6 +185,11 @@ func ExecuteMigration(ctx context.Context, l *zap.Logger, executor MigrationExec
 
 // beginMigration checks before executing migration and inserts a migration history record with pending status.
 func beginMigration(ctx context.Context, executor MigrationExecutor, m *db.MigrationInfo, prevSchema string, statement string) (insertedID int64, err error) {
+	// Convert verion to stored version.
+	storedVersion, err := toStoredVersion(m.UseSemanticVersion, m.Version, m.SemanticVersionSuffix)
```

Comment:
My feeling is it's less error-prone to do it when constructing the migration info at [ParseMigrationInfo](https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/plugin/db/driver.go?L241)

I tried to remove the version string and you can see the places where we need the change.

<img width="827" alt="CleanShot 2022-03-29 at 14 39 23@2x" src="https://user-images.githubusercontent.com/230323/160549343-55fc8f02-4bec-4e60-8704-0d1c5bb136e6.png">

Reply:
Yeah, I thought about that option while 
1) I feel like UseSemantic should be a project-level flag.
2) We want driver.ExecuteMigration() to have explicit semantic/non-semantic options since that's used by library.

Reply:
Btw, since this will be a breaking change, existing migration history will cause failures. I will add a version safe-guard around to/fromStoredVersion if we think the structure looks good.

Reply:
> Yeah, I thought about that option while
> 
> 1. I feel like UseSemantic should be a project-level flag.
> 2. We want driver.ExecuteMigration() to have explicit semantic/non-semantic options since that's used by library.

Both are true, but how would that affect putting storedVersion field inside MigrationInfo struct? What I meant is to keep both the original version and the storedVersion in MigrationInfo

Reply:
I might prefer keeping one since it's not easy to keep them in sync. We can always construct one from the other whenever needed. MigrationInfo is the API for setting migration, and users may not care about the storage format of the migration version as  implementation details.